src/characters/cbd-recipient.ts

```diff
@@ -124,9 +120,14 @@ export class CbdTDecDecrypter {
     variant: number,
     expectedRitualId: number
   ) {
-    const decryptedResponses = Object.entries(encryptedResponses).map(
-      ([ursula, encryptedResponse]) =>
-        encryptedResponse.decrypt(sessionSharedSecret[ursula])
+    const decryptedResponses = Object.entries(sessionSharedSecret).map(
+      ([ursula, sharedSecret]) => {
+        const encryptedResponse = encryptedResponses[ursula];
+        if (!encryptedResponse) {
+          throw new Error(`Missing encrypted response from ${ursula}`);
+        }
```

Comment:
Would this fail if any ursula did not return an encrypted response...? i.e.assuming a threshold number of responses have been returned, we only care about the ursulas that responded.

Reply:
The issue here is that Porter only waits for a threshold number of responses, so some Ursulas may not have responded by the time Porter returns because the threhold was already met. So you likely will have Ursulas without responses, but that not be a problem.